backend/lcfs/web/api/fuel_code/views.py

```diff
@@ -158,46 +156,47 @@ async def get_fuel_code(
     return await service.get_fuel_code(fuel_code_id)
 
 
+@router.post("/{fuel_code_id}/approve", status_code=status.HTTP_200_OK)
+@view_handler([RoleEnum.GOVERNMENT])
+async def approve_fuel_code(
+    request: Request,
+    fuel_code_id: int,
+    service: FuelCodeServices = Depends(),
+):
+    # TODO: Add Logic that checks if the code has necessary fields for approval
+    return await service.approve_fuel_code(fuel_code_id)
+
+
 @router.put("/{fuel_code_id}", status_code=status.HTTP_200_OK)
-@view_handler(["*"])
+@view_handler([RoleEnum.ANALYST])
 async def update_fuel_code(
     request: Request,
     fuel_code_id: int,
-    fuel_code_data: FuelCodeCreateSchema,
+    fuel_code_data: FuelCodeCreateUpdateSchema,
     service: FuelCodeServices = Depends(),
 ):
     return await service.update_fuel_code(fuel_code_id, fuel_code_data)
 
 
 @router.delete("/{fuel_code_id}", status_code=status.HTTP_200_OK)
-@view_handler(["*"])
+@view_handler([RoleEnum.ANALYST])
 async def delete_fuel_code(
     request: Request, fuel_code_id: int, service: FuelCodeServices = Depends()
 ):
     return await service.delete_fuel_code(fuel_code_id)
 
 
 @router.post(
-    "/save",
+    "/",
     response_model=Union[FuelCodeSchema, DeleteFuelCodeResponseSchema],
     status_code=status.HTTP_200_OK,
 )
-@view_handler([RoleEnum.ADMINISTRATOR])
-async def save_fuel_code_row(
+@view_handler([RoleEnum.ANALYST])
+async def create_fuel_code(
     request: Request,
-    request_data: FuelCodeCreateSchema = Body(...),
+    request_data: FuelCodeCreateUpdateSchema = Body(...),
     service: FuelCodeServices = Depends(),
 ):
-    """Endpoint to save a single fuel code row"""
-    fuel_code_id: Optional[int] = request_data.fuel_code_id
-
-    if request_data.deleted:
-        # Delete existing fuel code
-        await service.delete_fuel_code(fuel_code_id)
-        return DeleteFuelCodeResponseSchema(message="Fuel code deleted successfully")
-    elif fuel_code_id:
-        # Update existing fuel code
-        return await service.update_fuel_code(fuel_code_id, request_data)
-    else:
-        # Create new fuel code
-        return await service.create_fuel_code(request_data)
```

Comment:
Why you are not following the standard in the code to delete/update/create? Is just this functionality going to be different with that respect? Are we getting benefits to do so?

Reply:
I wasn't aware there was a standard, I was just following REST as I thought that's what we used. What should I be doing?

Reply:
Updated to combine update/save back into a single endpoint.